var/spack/repos/builtin/packages/hipsycl/package.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+from os import path
+from llnl.util import filesystem
+
+
+class Hipsycl(CMakePackage):
+    """hipSYCL is an implementation of the SYCL standard programming model
+    over NVIDIA CUDA/AMD HIP"""
+
+    homepage = "https://github.com/illuhad/hipSYCL"
+    url = "https://github.com/illuhad/hipSYCL/archive/v0.8.0.tar.gz"
+    git = "https://github.com/illuhad/hipSYCL.git"
```

Comment:
I think due to way that github packages source releases, it may be possible that the hipCPU submodule is missing from the tar file, in which case the CPU backend will not compile. Will the spack git clone also clone submodules?

Reply:
You were absolutely right, both the tarball and the `master` versions lacked `hipCPU` submodule. Addressed both of them, we now pull all versions from git with `--recurse-submodules`.